processing/src/main/java/org/apache/druid/segment/StringDimensionIndexer.java

```diff
@@ -308,10 +308,14 @@ public long estimateEncodedKeyComponentSize(int[] key)
     // even though they are stored just once. It may overestimate the size by a bit, but we wanted to leave
     // more buffer to be safe
     long estimatedSize = key.length * Integer.BYTES;
-    estimatedSize += Arrays.stream(key)
-                           .filter(element -> dimLookup.getValue(element) != null)
-                           .mapToLong(element -> dimLookup.getValue(element).length() * Character.BYTES)
-                           .sum();
+    long totalChars = 0;
+    for (int element : key) {
+      String val = dimLookup.getValue(element);
+      if (val != null) {
+        totalChars += val.length();
+      }
+    }
+    estimatedSize += totalChars * Character.BYTES;
```

Comment:
seeing second example , in quick succession, of stream slowness ( another one https://github.com/apache/incubator-druid/pull/8373 ) .
there are numerous discussions of streams being slower than regular for-loops in most cases on the internet.

using streams in hot paths looks like a bad idea, at least in java8 .